### DISCUSSION TITLE
Lost values should give a warning

### DISCUSSION THREAD

**Discussion by oliverlevay** [scala/scala3#17828] [Feature Requests]
```scala
case class KeyControl(up: String, left: String, down: String, right: String):
  def direction(key: String): (Int, Int) =
    if (key == left) then (-1, 0)
    if (key == right) then (1, 0)
    if (key == up) then (0, -1)
    if (key == down) then (0, 1)
    else (0, 0)
```
This code has lost values that are unreachable. It would be awesome if this fact was warned about to the developer.

**Comment by som-snytt** [comment]
It will try to warn about non-side-effecting expressions:
```
10 |    if (key == left) then 0
   |                          ^
   |                          A pure expression does nothing in statement position; you may be omitting necessary parentheses
```
The issue seems to be that tuple construction is not detected as "pure (for purposes of error reporting)".

Is that a bug or a feature request?

**Comment by som-snytt** [comment]
In Scala 3, one-armed if is typed as `Unit` (and not as though `else ()` as in Scala 2), so that `-Wvalue-discard` alerts for these cases.

In Scala 2, it warns under `-Wnonunit-statement`.
